cg/meta/deliver_ticket.py

```diff
@@ -22,108 +22,138 @@ def __init__(self, config: CGConfig):
         super().__init__(config)
         self.delivery_path: str = config.delivery_path
 
-    def get_all_cases_from_ticket(self, ticket_id: int) -> list:
+    def get_all_cases_from_ticket(self, ticket_id: int) -> List[models.Family]:
         cases: List[models.Family] = self.status_db.get_cases_from_ticket(ticket_id=ticket_id).all()
         return cases
 
-    def get_inbox_path(self, ticket_id: int) -> str:
-        cases = self.get_all_cases_from_ticket(ticket_id=ticket_id)
+    def get_inbox_path(self, ticket_id: int) -> Path:
+        cases: List[models.Family] = self.get_all_cases_from_ticket(ticket_id=ticket_id)
         if not cases:
-            LOG.warning("Could not find any cases for ticket_id %s", ticket_id)
+            LOG.warning(
+                "The customer id was not identified since no cases for ticket_id %s was found",
+                ticket_id,
+            )
             raise CgError()
 
         customer_id: str = cases[0].customer.internal_id
-        customer_inbox: str = (
-            Path(self.delivery_path, customer_id, "inbox", str(ticket_id)).as_posix() + "/"
-        )
+        customer_inbox = Path(self.delivery_path, customer_id, "inbox", str(ticket_id))
         return customer_inbox
 
     def check_if_upload_is_needed(self, ticket_id: int) -> bool:
         is_upload_needed = True
-        customer_inbox = self.get_inbox_path(ticket_id=ticket_id)
+        customer_inbox: Path = self.get_inbox_path(ticket_id=ticket_id)
         LOG.info("Checking if path exist: %s", customer_inbox)
-        if os.path.exists(customer_inbox):
+        if customer_inbox.exists():
             LOG.info("Could find path: %s", customer_inbox)
             is_upload_needed = False
         else:
             LOG.info("Could not find path: %s", customer_inbox)
         return is_upload_needed
 
     def generate_date_tag(self, ticket_id: int) -> str:
-        cases = self.get_all_cases_from_ticket(ticket_id=ticket_id)
+        cases: List[models.Family] = self.get_all_cases_from_ticket(ticket_id=ticket_id)
         date = str(cases[0].ordered_at)
         split_date = date.split(" ")
         return split_date[0]
 
+    def generate_output_filename(
+        self, dir_path: Path, date: str, dir_name: Path, read_direction: int
+    ) -> Path:
+        fastq_file_name = str(os.path.basename(dir_name)) + "_" + str(read_direction) + ".fastq.gz"
```

Comment:
What is the difference of `dir_path` and `dir_name` here`?
If you have the path to a file in Path format it is very easy to get parts of it:
```python
In [2]: from pathlib import Path

In [3]: a = Path("/Users/mans.magnusson/PycharmProjects/cg/README.md")

In [4]: a.parent
Out[4]: PosixPath('/Users/mans.magnusson/PycharmProjects/cg')

In [5]: a.parent.name
Out[5]: 'cg'

In [6]: a.name
Out[6]: 'README.md'
```
Hope this helps otherwise slack me and I'll help you

Reply:
with os this made sense, and they where different since one was the name of the folder, and the other was an absolute path. But yes, now I think they might be the same